pulp_container/app/models.py

```diff
@@ -452,6 +460,71 @@ class Meta:
         default_related_name = "%(app_label)s_%(model_name)s"
 
 
+class ManifestSigningService(SigningService):
+    """
+    Signing service used for creating container signatures.
+    """
+
+    def validate(self):
+        """
+        Validate a signing service for a container signature.
+
+        The validation seeks to ensure that the sign() method returns a dict as follows:
+
+        {"signature": "$SIG_PATH"}
+
+        The method creates a test image manifest, signs its manifest.json, and checks if the
+        signature can be verified by the provided public key.
+
+        Raises:
+            RuntimeError: If the validation has failed.
+
+        """
+        test_manifest = {
+            "schemaVersion": 2,
+            "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
+            "config": {
+                "mediaType": "application/vnd.docker.container.image.v1+json",
+                "size": 1456,
+                "digest": "sha256:7138284460ffa3bb6ee087344f5b051468b3f8697e2d1427bac1a208d4168123",
+            },
+            "layers": [
+                {
+                    "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
+                    "size": 772792,
+                    "digest": "sha256:e685c5c858e36338a47c627763b50dfe6035b547f1f75f0d39753d4e3121",
+                }
+            ],
+        }
+
+        with tempfile.TemporaryDirectory() as temp_directory_name:
+            manifest_file = tempfile.NamedTemporaryFile(dir=temp_directory_name, delete=False)
+            with open(manifest_file.name, "w") as outfile:
+                json.dump(test_manifest, outfile)
+            sig_path = os.path.join(temp_directory_name, "signature")
+
+            signed = self.sign(
+                manifest_file.name, env_vars={"REFERENCE": "test", "SIG_PATH": sig_path}
+            )
+
+            with open(signed["signature_path"], "rb") as fp:
+                verified = gpg.verify_file(fp)
+                if verified.trust_level is None or verified.trust_level < verified.TRUST_FULLY:
+                    raise RuntimeError(
+                        _(
+                            "The signature could not be verified or the trust level is too "
+                            "low. The signing script may generate invalid signatures."
+                        )
+                    )
+                elif verified.pubkey_fingerprint != self.pubkey_fingerprint:
+                    raise RuntimeError(
+                        _(
+                            "Fingerprints of the provided public key and the verified public "
+                            "key are not equal. The signing script is probably not valid."
```

Comment:
```suggestion
                            "key are not equal. The signing script is not valid."
```
We refuse to save it, so I'd say it _is_ invalid. (Same goes for the "may" above.)

Reply:
it's copy paste from core, should i also update wording in core?:)

Reply:
Actually, I'm taking it back. The signing service is invalid. And this comment tried to imply the script may be the reason. Please don't change it.